batchflow/models/torch/base.py

```diff
@@ -1847,3 +1853,87 @@ def _parse_profilers(self):
+
+
+    # Utilities for activations
+    @staticmethod
+    def get_blocks_and_activations(model, modules=None):
+        """ Retrieve intermediate blocks of the neural network model
+        and corresponding activation names.
+
+        Parameters
+        ----------
+        model : Network
+
+        modules : str or list of str, default None
+            The modules of the model for retrieving activations from.
+            If None, all of the parts from the `model.config['order']` will be used.
+
+        Returns
+        -------
+        activation_blocks : list of str
+            Intermediate activation blocks.
+        activation_names : list of str
+            Names of activations corresponding to activation blocks.
+
+        Examples
+        --------
+        >>> get_blocks_and_activations(model, modules=['encoder', 'embedding', 'decoder'])
+        >>> print(activation_blocks)
+        ['model.encoder["block-0"]', 'model.encoder["block-1"]', 'model.embedding', 'model.decoder["block-0"]']
+        >>> print(activation_names)
+        ['encoder_0', 'encoder_1', 'embedding_0', 'decoder_0']
+        """
+        modules = [modules] if isinstance(modules, str) else modules or model.config['order']
+        activation_names = []
+        activation_blocks = []
+
+        for module in modules:
+            number = -1
+            extracted_module = getattr(model, module)
+            if isinstance(extracted_module, Iterable):
+                for block in extracted_module:
+                    if 'block' in block:
+                        number += 1
+                        activation_blocks.append(f'model.{module}["{block}"]')
+                        activation_names.append(f'{module}_{number}')
+            else:
+                activation_blocks.append(f'model.{module}')
+                activation_names.append(f'{module}_0')
+
+        return activation_blocks, activation_names
+
+    @staticmethod
+    def reduce_channels(images, normalize=True, pca_instance=None):
+        """ Convert multichannel images to RGB images using PCA.
+
+        Parameters
+        ----------
+        images : np.ndarray
+            4D array of shape (B, C, H, W).
+        normalize : bool, default True
+            Whether to apply 'minmax' normalization to the images.
+        pca_instance : sklearn.decomposition.PCA, default None
+            Instance of the PCA.
+            If None, the PCA with n_components == 3 will be used.
+
+        Returns
+        -------
+        compressed_images : np.ndarray
+            Compressed RGB images of shape (B, H, W, 3).
+        explained_variance_ratio : np.ndarray
+            Percentage of variance explained by each of the selected components.
+        """
+        images = images.transpose(0, 2, 3, 1)
+        if pca_instance is None:
+            pca_instance = PCA(n_components=3)
+
+        compressed_images = pca_instance.fit_transform(images.reshape(-1, images.shape[-1]))
```

Comment:
Is there a difference between re-creating instance each time or using passed one? Is it even possible to continuously `fit` PCA?